src/coreclr/debug/di/process.cpp

```diff
@@ -11152,7 +11155,41 @@ void CordbProcess::FilterClrNotification(
     }
 }
 
+#if defined(TARGET_WINDOWS) && defined(TARGET_AMD64)
+void CordbProcess::HandleSetThreadContextNeeded(DWORD dwThreadId)
+{
+    LOG((LF_CORDB, LL_INFO10000, "RS HandleSetThreadContextNeeded\n"));
+
+    CONTEXT context = { 0 };
+    context.ContextFlags = CONTEXT_FULL;
+
+    GetLiveContext(dwThreadId, &context);
+    TADDR lsContextAddr = (TADDR)context.Rcx;
+    DWORD contextSize = (DWORD)context.Rdx;
+
+    if (contextSize == 0 || contextSize > sizeof(CONTEXT) + 25000)
+    {
+        _ASSERTE(!"Corrupted HandleSetThreadContextNeeded message received");
+
+        LOG((LF_CORDB, LL_INFO10000, "RS HandleSetThreadContextNeeded - Corrupted HandleSetThreadContextNeeded message received\n"));
+
+        ThrowHR(E_UNEXPECTED);
+    }
+
+    PCONTEXT pContext = (PCONTEXT)_alloca(contextSize);
+    HRESULT hr = GetDAC()->ReadData(lsContextAddr, contextSize, (BYTE*)pContext);
+    if (FAILED(hr))
+    {
+        _ASSERTE(!"ReadData failed");
 
+        LOG((LF_CORDB, LL_INFO10000, "RS HandleSetThreadContextNeeded - Unexpected result from ReadData (error: 0x%X).\n", hr));
+
+        ThrowHR(hr);
+    }
+
+    SetLiveContext(dwThreadId, pContext);
```

Comment:
> Instead of a new function this can GetDataTarget()->SetThreadContext(...)

Good suggestion.  Our implementation of ICorDebugMutableDataTarget only handles the legacy context scenario.  We either need to change it or add our own SetThreadContext.  I opted to create our own as I wanted minimize risk of the change and not alter existing debugging code (since we are close to the end of .NET 7).  I'll create a tracking item for .NET 8 to unify these code paths.

Reply:
In that case I'd suggest adding a comment in the code describing what is different between the new function and the data target variant and that we'd like to fold them back together again. I expect the most likely result of a 'refactor this later' work item is that it will be punted to Future and closed without ever doing it, but at least with the comment future folks will know whats up and they might do the refactoring if it is convenient.